### ISSUE TITLE
Mosquitto Will Not Bind to Both IPv4 and IPv6 for Websockets.

### ISSUE THREAD

**Issue by arshamahmadi** [eclipse/mosquitto#2628]
Hi, there is a similar issue reported under #1974 but it's marked as "resolved" by the bot. I believe this hasn't been resolved yet. Adding `socket_domain ipv4` to `/etc/mosquitto/conf.d/default.conf` will only switch from IPv6 to IPv4 but how can one have Mosquitto bind to both IPv4 and IPv6 just like how the MQTT protocol can bind to both IPv4 and IPv6 without any directives?

I can reproduce this in Mosquitto version 2.0.11 running on Ubuntu 22.04.1 LTS.

Thanks in advance for your help.

**Comment by ralight** [comment]
I believe it depends on how libwebsockets is compiled. It does not have support for IPv6 compiled in by default.

**Comment by cateee** [comment]
@ralight : no. In my case, but also in the original problem, the IPv6 web-socket is selected by default, only with explicit configuration `socket_domain ipv4` we see it as IPv4. Duplicating the listener changing the socket_domains is not working (configuration error), but if I select a different port, I can get both IPv4 and IPv6 websockets. Note TCP MQTT has not such problem: by default it opens both protocols.

**Comment by mjcross** [comment]
As of version 2.0.21 and OpenBSD 7.7 I can't get IPv6 websocket listeners to work - apparently this is due to issues with _libwebsockets_ and should be [fixed in v2.1 with built-in websockets support](https://github.com/eclipse-mosquitto/mosquitto/issues/3383#issuecomment-3340530531).

In the meantime as a workaround I'm using [websocketd](http://websocketd.com/) with nc (netcat) to expose the 'plain' MQTT socket as a websocket endpoint on IPv6:
```
$ websocketd -binary -port 8080 nc localhost 1883
```
This works fine and seems pretty straightforward.

**Comment by ralight** [comment]
2.1 is now available, so I am closing this issue.
